src/Components/Facility/FacilityCreate.tsx

```diff
@@ -243,6 +244,34 @@ export const FacilityCreate = (props: FacilityProps) => {
     },
   );
 
+  const facilitiesQuery = useQuery(routes.getAllFacilities);
+  const hubsQuery = useQuery(routes.getFacilityHubs, {
+    pathParams: {
+      id: facilityId!,
+    },
+    prefetch: !!facilityId,
+    onResponse: (res) => {
+      setHubFacilities(res.data?.results.map((d) => d.hub.id) as string[]);
+    },
+  });
```

Comment:
Avoid using additional states when using useQuery. 

```suggestion
  });
  const hubFacilities = hubsQuery.data?.results.map((d) => d.hub.id);
```